content/en/docs/components/pipelines/v1/sdk/connect-api.md

```diff
@@ -152,146 +152,186 @@ and give them access to submit pipelines in their own namespace.
                 )
+            dex_login_url = resp.url
 
-            # set the login url
-            auth_session["dex_login_url"] = resp.url
-
-        ################
-        # Attempt Dex Login
-        ################
+        # attempt Dex login
         resp = s.post(
-            auth_session["dex_login_url"],
-            data={"login": username, "password": password},
-            allow_redirects=True
+            dex_login_url,
+            data={"login": self._dex_username, "password": self._dex_password},
+            allow_redirects=True,
         )
+        if resp.status_code != 200:
+            raise RuntimeError(
+                f"HTTP status code '{resp.status_code}' for POST against: {dex_login_url}"
+            )
+
+        # if we were NOT redirected, then the login credentials were probably invalid
         if len(resp.history) == 0:
             raise RuntimeError(
-                f"Login credentials were probably invalid - "
-                f"No redirect after POST to: {auth_session['dex_login_url']}"
+                f"Login credentials are probably invalid - "
+                f"No redirect after POST to: {dex_login_url}"
             )
 
-        # store the session cookies in a "key1=value1; key2=value2" string
-        auth_session["session_cookie"] = "; ".join([f"{c.name}={c.value}" for c in s.cookies])
+        return "; ".join([f"{c.name}={c.value}" for c in s.cookies])
 
-    return auth_session
-```
+    def _create_kfp_client(self) -> kfp.Client:
+        try:
+            session_cookies = self._get_session_cookies()
+        except Exception as ex:
+            raise RuntimeError(f"Failed to get Dex session cookies") from ex
 
-__Step 3:__ this Python code uses the above `get_istio_auth_session()` function to create a `kfp.Client()`
+        # monkey patch the kfp.Client to support disabling SSL verification
+        # kfp only added support in v2: https://github.com/kubeflow/pipelines/pull/7174
+        original_load_config = kfp.Client._load_config
 
-```python
-import kfp
+        def patched_load_config(client_self, *args, **kwargs):
+            config = original_load_config(client_self, *args, **kwargs)
+            config.verify_ssl = not self._skip_tls_verify
+            return config
 
-KUBEFLOW_ENDPOINT = "http://localhost:8080"
-KUBEFLOW_USERNAME = "user@example.com"
-KUBEFLOW_PASSWORD = "12341234"
+        patched_kfp_client = kfp.Client
+        patched_kfp_client._load_config = patched_load_config
 
-auth_session = get_istio_auth_session(
-    url=KUBEFLOW_ENDPOINT,
-    username=KUBEFLOW_USERNAME,
-    password=KUBEFLOW_PASSWORD
+        return patched_kfp_client(
+            host=self._api_url,
+            cookies=session_cookies,
+        )
+
+    def get_kfp_client(self) -> kfp.Client:
+        """Get a newly authenticated Kubeflow Pipelines client."""
+        return self._create_kfp_client()
```

Comment:
NIT: Looks like this func is only calling `_create_kfp_client()`, and the func name `get_kfp_client()` which suggests that it is retrieving the same kfp.Client is actually generating new session cookie and a new kfp client every time being called. Maybe consider making `_create_kfp_client()` public directly?

Reply:
I agree that `get_kfp_client` could be read either way, which is why the docs say "newly authenticated" and there is a tip in the following example which says `TIP: long-lived sessions might need to get a new client when their session expires`.

For reference, I think the class is more readable with a single (clearly public) method that is intended for the user.

However, could rename the current `get_kfp_client()` to `create_kfp_client()` but I actually think that is misleading, I will leave it up to you to decide.

Reply:
It seems that renaming to `create_kfp_client()` is clear to me. Would you like to explain why is it misleading?

Reply:
@zijianjoy I agree, let's rename, I have done this in https://github.com/kubeflow/website/pull/3565/commits/c930d58164810ef67fcf5ed638028e84f179c225